client/components/new-post.jsx

```diff
@@ -0,0 +1,271 @@
+import React from 'react';
+import debounce from '../lib/debounce';
+
+export default class NewPostForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalView: 'none',
+      gameList: [],
+      gameSearch: '',
+      searchStatus: 'empty',
+      chosenGame: null
+    };
+    this.handleSearch = this.handleSearch.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.runSearch = debounce(this.runSearch.bind(this), 500);
+    this.handleType = this.handleType.bind(this);
+    this.renderSearch = this.renderSearch.bind(this);
+    this.handlePost = this.handlePost.bind(this);
+  }
+
+  handleSearch(event) {
+    const chosenGame = this.state.gameList.find(gameObject => gameObject.gameId === event.target.dataset.gameid);
+    fetch(`/api/boardGameInfo/${chosenGame.gameId}`)
+      .then(response => response.json())
+      .then(gameInfo => {
+        gameInfo.name = chosenGame.name;
+        gameInfo.gameId = chosenGame.gameId;
+        this.setState({ chosenGame: gameInfo, searchStatus: 'empty', modalView: 'none' });
+      })
+      .then(() => {
+        this.setState({ searchStatus: 'chosen' });
+      });
+  }
+
+  handleClick() {
+    this.setState({ modalView: 'search' });
+    this.runSearch();
+  }
+
+  handleType(event) {
+    if (event.target.value !== this.state.gameSearch) {
+      this.setState({ gameSearch: event.target.value }, this.runSearch);
+    }
+  }
+
+  runSearch() {
+    if (this.state.gameSearch.length > 0) {
+      this.setState({ searchStatus: 'searching' });
+      const currentSearch = this.activeSearch = fetch(`/api/boardGames/${this.state.gameSearch}`)
+        .then(response => response.json())
+        .then(games => {
+          if (this.activeSearch !== currentSearch) return;
+          if (games.error || games.length < 1) {
+            this.setState({ searchStatus: 'none' });
+          } else {
+            this.setState({ gameList: games, searchStatus: 'result' });
+          }
+        });
+    } else {
+      this.setState({ searchStatus: 'empty' });
+    }
+  }
+
+  handlePost(event) {
+    event.preventDefault();
+    const header = new Headers();
+    header.append('Content-Type', 'application/json');
+    let body = {
+      comments: event.target.elements.comments.value,
+      gameId: this.state.chosenGame.gameId,
+      gameImg: this.state.chosenGame.imageUrl,
+      game: event.target.elements.game.value,
+      lender: event.target.elements.name.value
+    };
+    body = JSON.stringify(body);
+    const init = {
+      method: 'POST',
+      headers: header,
+      body: body
+    };
+    fetch('/api/boardGamePosts', init)
+      .then(response => response.json())
+      .then(post => {
+        event.target.elements.name.value = '';
+        event.target.elements.game.value = '';
+        event.target.elements.comments.value = '';
```

Comment:
we don't mutate the dom like this in react. instead, `setState` to reset the form input values.